cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/MeasureDefBuilderTest.java

```diff
@@ -0,0 +1,457 @@
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.opencds.cqf.fhir.cr.measure.constant.MeasureReportConstants.IMPROVEMENT_NOTATION_SYSTEM_DECREASE;
+import static org.opencds.cqf.fhir.cr.measure.constant.MeasureReportConstants.IMPROVEMENT_NOTATION_SYSTEM_INCREASE;
+import static org.opencds.cqf.fhir.cr.measure.constant.MeasureReportConstants.MEASUREREPORT_IMPROVEMENT_NOTATION_EXTENSION;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.parser.IParser;
+import org.hl7.fhir.exceptions.FHIRException;
+import org.hl7.fhir.r4.model.CodeType;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Extension;
+import org.junit.jupiter.api.Test;
+import org.opencds.cqf.fhir.cr.measure.common.MeasureDef;
+import org.opencds.cqf.fhir.cr.measure.common.MeasureScoring;
+import org.opencds.cqf.fhir.cr.measure.constant.MeasureConstants;
+import org.opencds.cqf.fhir.cr.measure.constant.MeasureReportConstants;
+
+/**
+ * Test MeasureDefBuilder on different scenarios around group level Measure settings and if they are properly being set
+ */
+public class MeasureDefBuilderTest {
+    private final CodeableConcept increase = new CodeableConcept(new Coding(
+            MeasureReportConstants.MEASUREREPORT_IMPROVEMENT_NOTATION_SYSTEM,
+            IMPROVEMENT_NOTATION_SYSTEM_INCREASE,
+            MeasureReportConstants.IMPROVEMENT_NOTATION_SYSTEM_INCREASE_DISPLAY));
+
+    private final CodeableConcept decrease = new CodeableConcept(new Coding(
+            MeasureReportConstants.MEASUREREPORT_IMPROVEMENT_NOTATION_SYSTEM,
+            IMPROVEMENT_NOTATION_SYSTEM_DECREASE,
+            MeasureReportConstants.IMPROVEMENT_NOTATION_SYSTEM_DECREASE_DISPLAY));
+    private final CodeableConcept invalid = new CodeableConcept(
+            new Coding(MeasureReportConstants.MEASUREREPORT_IMPROVEMENT_NOTATION_SYSTEM, "fake", "Fake"));
+    private final FhirContext fhirContext = FhirContext.forCached(FhirVersionEnum.R4);
+    private final IParser parser = fhirContext.newJsonParser();
+
+    public MeasureDef measureDefBuilder(
+            String group1Basis,
+            String group1Scoring,
+            CodeableConcept group1ImpNotation,
+            String group2Basis,
+            String group2Scoring,
+            CodeableConcept group2ImpNotation,
+            String measureBasis,
+            String measureScoring,
+            CodeableConcept measureImpNotation) {
+
+        R4MeasureDefBuilder defBuilder = new R4MeasureDefBuilder();
+        org.hl7.fhir.r4.model.Measure measure = (org.hl7.fhir.r4.model.Measure)
+                parser.parseResource(MeasureDefBuilderTest.class.getResourceAsStream("TemplateMeasure.json"));
+
+        var group1 = measure.getGroup().stream()
+                .filter(t -> t.getId().equals("group-1"))
+                .findFirst()
+                .get();
```

Comment:
Suggestion:  orElseThrow or some clear signal if the groups are empty

Reply:
this is a static template for this test, so no risk of not finding group